src/server/saved_objects/client/lib/create_find_query.js

```diff
@@ -35,5 +40,20 @@ export function createFindQuery(options = {}) {
     });
   }
 
-  return { version: true, query: { bool } };
+  const query = { version: true, query: { bool } };
+
+  if (sortField) {
+    const value = {
+      order: sortOrder,
+      unmapped_type: get(mappings, `${type}.properties.${sortField}.type`)
```

Comment:
Not currently because of https://www.elastic.co/guide/en/elasticsearch/reference/2.4/dots-in-names.html . However, I do see the benefit of not having to manually create this path string and prefer the approach that you've recommended as it's safe if Elasticsearch does decide to allow dots in field names.